### ISSUE TITLE
Get line number of classes and functions...

### ISSUE THREAD

**Issue by jedie** [mitmproxy/pdoc#268]
I implement a README generate using pdoc here: https://github.com/boxine/bx_py_utils/pull/76
Example result is currently: https://github.com/boxine/bx_py_utils/blob/auto-doc/README.md

The idea is to add links to the github code view page, e.g.: 
https://github.com/boxine/bx_py_utils/blob/auto-doc/bx_py_utils/auto_doc.py#L18

But I can't find a way to get the line number of the functions and classes from pdoc.
In short i do:
```
module_obj = extract.load_module(module_name)
pdoc_module = Module(module_obj)

for item in pdoc_module.classes:
    # ...

for item in pdoc_module.functions
    # ...
```

In the end i process a list of `pdoc.doc.Class` and `pdoc.doc.Function` instances.
Is it possible to the the line numbers of these instances?!?

**Comment by mhils** [comment]
Would calling `inspect.getsourcelines(item.obj)` work for you? I think we could generally add a `pdoc.doc.Doc.sourcelines` property based on that if there is greater interest.

**Comment by jedie** [comment]
A yes `inspect.getsourcelines(item.obj)` works and i can use: `lines, lnum = inspect.findsource(pdoc_item.obj)` to get the line number. Great. Thanks.

Maybe i can add a PR here... But if this project doesn't need this information, it makes no sense isn't it?

**Comment by mhils** [comment]
Using pdoc as a library is absolutely an intended use case. :) I've opened #269 with a possible API! 😃

**Comment by jedie** [comment]
The generated README is online here: https://github.com/boxine/bx_py_utils#existing-stuff
And it looks like:

-----
![grafik](https://user-images.githubusercontent.com/71315/120209577-24b49d00-c22f-11eb-932d-e8e02cae0854.png)
-----

Another problematic part is to get the relativ path of the source file. I made a helper for this here:

https://github.com/boxine/bx_py_utils/blob/488efb9f9c4993ac97f46af29481bfb90b895251/bx_py_utils/auto_doc.py#L20 

But i think i missed something. Think pdoc should hold this path already, isn't it?

**Comment by mhils** [comment]
[`Doc.source_file`](https://pdoc.dev/docs/pdoc/doc.html#Doc.source_file) is probably what you are looking for?

**Comment by jedie** [comment]
No. With this I get the absolute file path. I need the relative path.

**Comment by mhils** [comment]
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to should have you covered :)

**Comment by jedie** [comment]
Yes, i use it in my solution, see above... The problem is: what's the base path of the project?

**Comment by mhils** [comment]
I'm not quite sure I understand what you want to achieve with the base path of your Python module here. Let's assume you want to document `bx_py_utils.iteration.chunk_iterable`. This function's `.modulename` will be `bx_py_utils.iteration`, so I would just construct a path like this:
```python
url = f"https://github.com/boxine/bx_py_utils/blob/master/{ x.modulename.replace('.', '/') }.py#L4"
```

Does that make sense?

**Comment by mhils** [comment]
The original issue has been fixed, so I'm closing this here! 😃
